pokemongo_bot/walkers/step_walker.py

```diff
@@ -8,147 +8,105 @@
-            # need to speed one even when teleporting small distances
-            sleep(1)  # sleep one second plus a random delta
-            return True
-
-
-        self._new_position = self._get_next_pos(self.initLat, self.initLng, self.bearing, self.speed, self.precision)
-        cAlt = self.initAlt + random_alt_delta()
-
-        self.api.set_position(self._new_position[0], self._new_position[1], cAlt)
-        self.bot.event_manager.emit(
-            'position_update',
-            sender=self,
-            level='debug',
-            data={
-                'current_position': (self._new_position[0], self._new_position[1], cAlt),
-                'last_position': (self.initLat, self.initLng, self.initAlt),
-                'distance': '',
-                'distance_unit': ''
-            }
-        )
         self.bot.heartbeat()
 
-        sleep(1)  # sleep one second plus a random delta
-        # self._work_at_position(
-        #     self.initLat, self.initLng,
-        #     alt, False)
+        return self.is_arrived()
+
+    def is_arrived(self):
+        return distance(self.bot.position[0], self.bot.position[1], self.dest_lat, self.dest_lng) <= self.precision
 
     def _calc_bearing(self, start_lat, start_lng, dest_lat, dest_lng):
         """
         Calculates the bearing between two points.
-    
+
         The formulae used is the following:
             θ = atan2(sin(Δlong).cos(lat2),
                       cos(lat1).sin(lat2) − sin(lat1).cos(lat2).cos(Δlong))
-    
+
         :Parameters:
           - `start_lat in decimal degrees
           - `start_lng in decimal degrees
           - `dest_lat in decimal degrees
           - `dest_lng in decimal degrees
-    
+
         :Returns:
           The bearing in degrees
-    
+
         :Returns Type:
           float
         """
-    
+
         lat1 = math.radians(start_lat)
         lat2 = math.radians(dest_lat)
-    
+
         diffLong = math.radians(dest_lng - start_lng)
-    
+
         x = math.sin(diffLong) * math.cos(lat2)
-        y = math.cos(lat1) * math.sin(lat2) - (math.sin(lat1)
-                * math.cos(lat2) * math.cos(diffLong))
-    
+        y = math.cos(lat1) * math.sin(lat2) - (math.sin(lat1) * math.cos(lat2) * math.cos(diffLong))
+
         initial_bearing = math.atan2(x, y)
-    
+
         # Now we have the initial bearing but math.atan2 return values
         # from -180° to + 180° which is not what we want for a compass bearing
         # The solution is to normalize the initial bearing as shown below
         initial_bearing = math.degrees(initial_bearing)
         compass_bearing = (initial_bearing + 360) % 360
-    
+
         return compass_bearing
-        
-    def _get_next_pos(self, lat, lon, bearing, speed, precision):
-        origin = Point(lat, lon)
+
+    def get_next_position(self, origin_lat, origin_lng, origin_alt, dest_lat, dest_lng, dest_alt, speed):
+        origin = Point(origin_lat, origin_lng, origin_alt)
```

Comment:
Point(origin_lat, origin_lng, origin_alt)

don't you want the point to be (origin_lat, origin_lng, dest_alt)??